### PR TITLE
Add clangd recommendation for VS Code

### PR DESCRIPTION
**Pull request by justindeguzman**
### Changelog category (leave one):

- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added ClangD recommendation for VSCode.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit ce7ee1868517802907d6f075f960ef6988adf1a2 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56071/ce7ee1868517802907d6f075f960ef6988adf1a2/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

**Review by nikitamikhaylov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by jsc0218** [comment]
I tried to use clangd since I use VSCode. Unfortunately, I didn't find it is very powerful. I would say clangd provides more colourful variables. But it fails to have proper "go to definition", (for example, "go to definition" on UUID, from vector<UUID>, it goes to vector definition instead). Another drawback is since clangd has conflicts with IntelliSense, I disables IntelliSense according to clangd's pop up info. But after uninstall clangd, my IntelliSense still doesn't work, which took me 2 hours to find it back.

**Comment by nikitamikhaylov** [comment]
@justindeguzman Style Check has failed.

File: docs/en/development/developer-instruction.md [hunk 1]
```diff
@@ -219,13 +219,21 @@ You can also run your custom-built ClickHouse binary with the config file from t
 
 ## IDE (Integrated Development Environment) {#ide-integrated-development-environment}
 
-If you do not know which IDE to use, we recommend that you use CLion. CLion is commercial software, but it offers 30 days free trial period. It is also free of charge for students. CLion can be used both on Linux and on macOS.
+**CLion (recommended)**
```
[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply docs/en/development/developer-instruction.md:222]: I use CLion - it is slow, memory-hungry, and has bugs, but code navigation works and overall ergonomics is good.
Let's make a poll across the core team - which IDE will be more used will be recommended.

**Pull request merged**